crypto/x509/x_pubkey.c

```diff
@@ -977,20 +977,25 @@ int ossl_i2d_X448_PUBKEY(const ECX_KEY *a, unsigned char **pp)
 
 #endif
 
+void X509_PUBKEY_set0_public_key(X509_PUBKEY *pub,
+                                 unsigned char *penc, int penclen)
+{
+    OPENSSL_free(pub->public_key->data);
```

Comment:
First part is equivalent to ASN1_STRING_set0(pub->public_key);

Reply:
Ah, right!
Added a refactoring commit that makes use `ASN1_STRING_set0()` here and other places in libcrypto.